dvc/remote/local/__init__.py

```diff
@@ -352,12 +359,21 @@ def status(self, checksum_infos, remote, jobs=None, show_checksums=False):
 
         progress.update_target(title, 30, 100)
 
+        local_exists = self.cache_exists(md5s)
+
+        # This is a performance optimization. We can safely assume that,
+        # if the resources that we want to fetch are already cached,
+        # there's no need to check the remote storage for the existance of
+        # those files.
+        if download and local_exists == md5s:
+            self._fill_statuses(ret, local_exists, local_exists)
+            progress.finish_target(title)
+            return ret
+
         remote_exists = list(remote.cache_exists(md5s))
 
         progress.update_target(title, 90, 100)
```

Comment:
Need to also update progress numbers so they match increments that were before :) It is kinda lame, but still :slightly_smiling_face:

Reply:
:thinking: but it would happen so fast that they wouldn't be perceived by the user
What is the reason behind this? `finish_target` updates the bar to a 100

Reply:
@mroutis it is a temporary progress bar until we start using it properly in `cache_exists`. It is used to provide at least some information for long-running steps in status checking.

Reply:
@efiop , just to be on the same page, are you suggesting something like:
```diff
        if download and local_exists == md5s:
            self._fill_statuses(ret, local_exists, local_exists)
+           progress.update_target(title, 90, 100)
            progress.finish_target(title)
            return ret
```

Reply:
See comment down below.